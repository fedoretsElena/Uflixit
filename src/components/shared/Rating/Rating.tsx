import React from "react";

interface IProps {
    value: number;
}

function getBgValue(value: number): string {
    if (value > 8) {
        return "success";
    } else if (value > 6) {
        return "warning";
    } else {
        return "secondary";
    }
}

const Rating = ({ value }: IProps) => (
    <span className={"badge badge-" + getBgValue(value)}>{value}</span>
);

export default Rating;
