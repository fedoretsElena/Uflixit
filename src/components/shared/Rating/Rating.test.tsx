import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";

import Rating, { getBgValue } from "./Rating";

describe("Rating component", () => {
    let component: any;

    beforeEach(() => {
        component = render(<Rating value={7} />);
    });

    afterEach(cleanup);

    it("should render", () => {
        const { asFragment } = component; // as HTML that was rendered

        expect(asFragment()).toMatchSnapshot();
    });

    it("should insert number inside span", () => {
        const { getByTestId } = component; // as HTML that was rendered

        expect(getByTestId("badge-container")).toHaveTextContent("7");
    });

    it("should have warning class name if 6 < value < 8", () => {
        const { getByTestId } = component; // as HTML that was rendered

        expect(getByTestId("badge-container")).toHaveClass("badge-warning");
    });
});

describe("getBgValue function", () => {
    it("should return success class if value > 8", () => {
        expect(getBgValue(9)).toEqual("success");
    });

    it("should return success class if 6 < value < 8", () => {
        expect(getBgValue(7)).toEqual("warning");
    });

    it("should return success class if value < 6", () => {
        expect(getBgValue(3.5)).toEqual("secondary");
    });
});
