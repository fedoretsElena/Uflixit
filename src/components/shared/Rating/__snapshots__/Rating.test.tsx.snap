// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component should render 1`] = `
<DocumentFragment>
  <span
    class="badge badge-warning"
    data-testid="badge-container"
  >
    7
  </span>
</DocumentFragment>
`;
